test/integration/smoke/test_dynamicroles.py

```diff
@@ -0,0 +1,474 @@
+                len(list_rolepermissions) > 0,
+                "List rolepermissions response was empty"
+            )
+
+
+    @attr(tags=['advanced', 'simulator', 'basic', 'sg'], required_hardware=False)
+    def test_rolepermission_lifecycle_create(self):
+        """
+            Tests creation of role permission
+        """
+        # Reuse self.rolepermission created in setUp()
+        try:
+            rolepermission = RolePermission.create(
+                self.apiclient,
+                self.testdata["rolepermission"]
+            )
+            self.fail("An exception was expected when creating duplicate role permissions")
+        except CloudstackAPIException: pass
+
+        list_rolepermissions = RolePermission.list(self.apiclient, roleid=self.role.id)
+        self.assertEqual(
+            isinstance(list_rolepermissions, list),
+            True,
+            "List rolepermissions response was not a valid list"
+        )
+        self.assertNotEqual(
+            len(list_rolepermissions),
+            0,
+            "List rolepermissions response was empty"
+        )
+        self.assertEqual(
+            list_rolepermissions[0].rule,
+            self.testdata["rolepermission"]["rule"],
+            msg="Role permission rule does not match the test data"
+        )
+        self.assertEqual(
+            list_rolepermissions[0].permission,
+            self.testdata["rolepermission"]["permission"],
+            msg="Role permission permission-type does not match the test data"
+        )
+
+
+    @attr(tags=['advanced', 'simulator', 'basic', 'sg'], required_hardware=False)
+    def test_rolepermission_lifecycle_update(self):
+        """
+            Tests updation of role permission
+        """
+        new_rule = "some*Rule" + self.getRandomString()
+        self.rolepermission.update(self.apiclient, rule=new_rule, permission='deny')
+        updated_rolepermission = filter(lambda p: p.id == self.rolepermission.id,
+                                        RolePermission.list(self.apiclient, roleid=self.role.id))[0]
+        self.assertEqual(
+            updated_rolepermission.rule,
+            new_rule,
+            msg="Role permission rule does not match updated role permission"
+        )
+        self.assertEqual(
+            updated_rolepermission.permission,
+            'deny',
+            msg="Role permission permission-type does not match updated role permission"
+        )
+
+
+    @attr(tags=['advanced', 'simulator', 'basic', 'sg'], required_hardware=False)
+    def test_rolepermission_lifecycle_update_invalid_rule(self):
+        """
+            Tests updation of role permission with an invalid rule
+        """
+        for badrule in [" ", "^delete.*$", "some-@p1-!"]:
+            try:
+                self.rolepermission.update(self.apiclient, rule=badrule, permission='deny')
+                self.fail("Invalid role permission rule got updated")
+            except CloudstackAPIException: pass
```

Comment:
We can do that but since we've catching any ACS server api exception caught by CloudStackAPIException, we know it's ACS specific API exception and not any HTTP exception.